src/shared/utils/__tests__/getComponentName-test.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var getComponentName = require('getComponentName');
+var React = require('React');
+
+describe('getComponentName', function() {
+  it('should prefer `displayName`', function() {
```

Comment:
Let’s structure it like this:

* `describe('getComponentName')`
  - `describe('stateless functional component')`
    * `it('should prefer displayName')`
    * `it('should use name when there is no displayName')`
    * `it('should fall back to ReactComponent')`
  - `describe('ES class')`
    * `it('should prefer displayName')`
    * `it('should use name when there is no displayName')`
    * `it('should fall back to ReactComponent')`
  - `describe('createClass')`
    * `it('should prefer displayName')`
    * `it('should fall back to ReactComponent')`

or something similar.